httomo/ui_layer.py

```diff
@@ -0,0 +1,118 @@
+import yaml
+from pathlib import Path
+from typing import Any, Dict, List, Protocol, TypeAlias
+from importlib import import_module, util
+import sys
+import os
+import re
+
+from mpi4py import MPI
+from mpi4py.MPI import Comm
+
+import httomo
+from httomo.logger import setup_logger
+
+from httomo.utils import log_exception
+from httomo.methods_database.query import MethodDatabaseRepository
+from httomo.runner.pipeline import Pipeline
+
+from httomo.runner.backend_wrapper import make_backend_wrapper
+from httomo.runner.loader import make_loader
+from httomo.runner.output_ref import OutputRef
+from httomo.runner.task_runner import TaskRunner
+
+from datetime import datetime
+
+MethodConfig: TypeAlias = Dict[str, Dict[str, Any]]
+PipelineStageConfig: TypeAlias = List[MethodConfig]
+PipelineConfig: TypeAlias = List[PipelineStageConfig]
```

Comment:
Might want to be a bit careful using these type aliases in terms of avoiding confusion, they were defined when there were "stages" in the pipeline (ie, "pre-processing stage", "main pipeline stage") which is now no longer the case, so they need some updating if they are kept.

In particular:
- the `PipelineStageConfig` type alias has a name which doesn't make sense in the context of stages now being removed from pipelines -> it should probably be renamed to `PipelineConfig`
- the `PipelineConfig` type alias is now irrelevant, since there's no need to have a type which is a list of `PipelineStageConfig` types